### PR TITLE
Rewrite and restructure API docs for V14

### PR DESCRIPTION
**Pull request by kjac**
## Description

A _lot_ of API controller related docs reference Umbraco core features that have either been obsoleted or completely removed in V14:

- The `UmbracoApiController` base class is obsolete; moving forward, public facing API controllers should be based on the ASP.NET Core `Controller` base.
- The `[PluginController]` annotation only works with `UmbracoApiController`, so that needs documenting as if it had been obsoleted (which it will be eventually, but that's another task).
- The `UmbracoAuthorizedApiController` and `UmbracoAuthorizedJsonController` base classes have been removed entirely; the `ManagementApiControllerBase` takes their place and is already well documented in [this tutorial](https://docs.umbraco.com/umbraco-cms/tutorials/creating-a-backoffice-api).

This PR attempts to clean up the docs. In this effort, certain bits of the existing documentation has been moved around. In particular, everything in [this article](https://docs.umbraco.com/umbraco-cms/v/13.latest-lts/reference/routing/umbraco-api-controllers/authorization) has been moved:

- The `MemberAuthorizeAttribute` docs has been moved to the main "Umbraco API Controllers" article (see the "Adding member protection to public APIs" section).
- The custom policies docs has been moved to a brand new part of the Management API tutorial ("Access policies").

Also worth noting - I have added a new article about porting over old API controllers. I realize this really shouldn't be part of the "current" docs... but given the quite big impact of removing or obsoleting those API controller base classes, I felt it was in order to have this article. 

## Type of suggestion

* [ ] Typo/grammar fix
* [x] Updated outdated content
* [x] New content
* [x] Updates related to a new version
* [ ] Other

## Product & version (if relevant)

V14

## Deadline (if relevant)

As soon as possible.

### CODE REVIEW

**Comment by eshanrnh** [comment]
Thanks, @kjac for the PR 🙌We will review it as soon as we can 💪

File: 14/umbraco-cms/reference/routing/umbraco-api-controllers/README.md [hunk 1]
```diff
@@ -1,196 +1,152 @@
...
-All locally declared Umbraco API controllers will be routed under the url path of:
-
-`~/Umbraco/Api/[YourControllerName]`
-
-E.g. \*`~/Umbraco/Api/Products/GetAllProducts`
+{% hint style="warning" %}
+In Umbraco 13 and below, the recommended approach was to base API controllers on the `UmbracoApiController` class. However, `UmbracoApiController` is obsolete in Umbraco 14 and will be removed in Umbraco 15.
 
-Note that the "Controller" part of your controller name gets stripped away.
+For more details, read [Porting old Umbraco APIs](authorization.md).
```
Comment (jonat123) [inline 14/umbraco-cms/reference/routing/umbraco-api-controllers/README.md:36]:
```suggestion
Read the article [Porting old Umbraco APIs](authorization.md) for more details.
```

File: 14/umbraco-cms/reference/routing/umbraco-api-controllers/porting-old-umbraco-apis.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+[Route("/umbraco/api/products")]
+public class ProductsController : Controller
+{
+    [HttpGet("getall")]
+    public IActionResult GetAll() => Ok(new[] { "Table", "Chair", "Desk", "Computer" });
+}
+```
+{% endcode %}
+
+...is routing wise equivalent to this `UmbracoApiController` implementation:
```
Comment (jonat123) [inline 14/umbraco-cms/reference/routing/umbraco-api-controllers/porting-old-umbraco-apis.md:33]:
```suggestion
...Is routing-wise equivalent to this `UmbracoApiController` implementation:
```

**Review by jonat123 — APPROVED** [review]
@kjac Had a couple of suggestions, however, they were quite minor so I have just merged them in :D 

Other than that, everything looks good, so will go ahead and merge in this PR 😄

**Pull request merged**